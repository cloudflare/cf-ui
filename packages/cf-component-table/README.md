# cf-component-table

> Cloudflare Table Component

## Installation

```sh
$ npm install cf-component-table
```

## Usage

```js
const React = require('react');
const {
  Table,
  TableHead,
  TableBody,
  TableFoot,
  TableRow,
  TableHeadCell,
  TableCell
} = require('../../src/index');

export default class Application extends React.Component {
  render() {
    const data = [
      [1, 2, 3],
      [4, 5, 6],
      [7, 8, 9]
    ];

    return (
      <div>
        <h1>Cloudflare Table Component Example</h1>
        <Table>
          <TableHead>
            <TableRow>
              <TableHeadCell>One</TableHeadCell>
              <TableHeadCell>Two</TableHeadCell>
              <TableHeadCell>Three</TableHeadCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {data.map(row =>
              <TableRow>
                {row.map(cell =>
                  <TableCell>{cell}</TableCell>
                )}
              </TableRow>
            )}
          </TableBody>
        </Table>
      </div>
    );
  }
}
```
