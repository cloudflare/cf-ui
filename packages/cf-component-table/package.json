{
  "name": "cf-component-table",
  "description": "Cloudflare Table Component",
  "version": "2.4.1",
  "main": "lib/index.js",
  "module": "es/index.js",
  "jsnext:main": "es/index.js",
  "author": "James Kyle <jkyle@cloudflare.com>",
  "license": "BSD-3-Clause",
  "publishConfig": {
    "registry": "http://registry.npmjs.org/"
  },
  "dependencies": {
    "cf-style-container": "^1.3.1",
    "prop-types": "^15.5.8"
  },
  "peerDependencies": {
    "react": "^0.14.9 || ^15.0.0-0 || ^16.0.0-0"
  },
  "devDependencies": {
    "cf-style-provider": "^2.0.0"
  }
}
