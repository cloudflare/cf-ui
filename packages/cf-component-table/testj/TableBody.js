import React from 'react';
import renderer from 'react-test-renderer';
import { TableBody } from 'cf-component-table';

describe('TableBody', () => {
  test('should render', () => {
    const component = renderer.create(<TableBody>TableBody</TableBody>);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('should render extra class names', () => {
    const component = renderer.create(
      <TableBody className="extra">TableBody</TableBody>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
