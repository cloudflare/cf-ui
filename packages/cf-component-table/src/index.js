import Table, { createTable } from './Table';
import TableBody, { createTableBody } from './TableBody';
import TableCell, { createTableCell } from './TableCell';
import TableFoot, { createTableFoot } from './TableFoot';
import TableHead, { createTableHead } from './TableHead';
import TableHeadCell, { createTableHeadCell } from './TableHeadCell';
import TableRow, { createTableRow } from './TableRow';
import TablePropTypes from './TablePropTypes';

export {
  Table,
  TableBody,
  TableCell,
  TableFoot,
  TableHead,
  TableHeadCell,
  TableRow,
  createTable,
  createTableBody,
  createTableCell,
  createTableFoot,
  createTableHead,
  createTableHeadCell,
  createTableRow,
  TablePropTypes
};
