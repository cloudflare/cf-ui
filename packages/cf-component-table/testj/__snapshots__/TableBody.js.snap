// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableBody should render 1`] = `
<tbody
  className="cf-table__body"
>
  TableBody
</tbody>
`;

exports[`TableBody should render extra class names 1`] = `
<tbody
  className="cf-table__body extra"
>
  TableBody
</tbody>
`;
